sqlmesh/core/selector.py

```diff
@@ -97,7 +113,67 @@ def select_models(
 
         return models
 
-    def expand_model_selections(self, model_selections: t.Iterable[str]) -> t.Set[str]:
+    @staticmethod
+    def _get_value_and_dependency_inclusion(value: str) -> t.Tuple[str, bool, bool]:
+        include_upstream = False
+        include_downstream = False
+        if value[0] == "+":
+            value = value[1:]
+            include_upstream = True
+        if value[-1] == "+":
+            value = value[:-1]
+            include_downstream = True
+        return value, include_upstream, include_downstream
+
+    def _get_models(
+        self, model_name: str, include_upstream: bool, include_downstream: bool
+    ) -> t.Set[str]:
+        result = {model_name}
+        if include_upstream:
+            result.update(self._dag.upstream(model_name))
+        if include_downstream:
+            result.update(self._dag.downstream(model_name))
+        return result
+
+    def _expand_model_tag(self, tag_selection: str) -> t.Set[str]:
+        """
+        Expands a set of model tags into a set of model names.
+        The tag matching is case-insensitive and supports wildcards and + prefix and suffix to
+        include upstream and downstream models.
+
+        Args:
+            tag_selection: A tag to match models against.
+
+        Returns:
+            A set of model names.
+        """
+        result = set()
+        matched_tags = set()
+        (
+            selection,
+            include_upstream,
+            include_downstream,
+        ) = self._get_value_and_dependency_inclusion(tag_selection.lower())
+
+        if "*" in selection:
+            for model_tag in self._models_by_tag:
+                if fnmatch.fnmatchcase(model_tag, selection):
+                    matched_tags.add(model_tag)
+        elif selection in self._models_by_tag:
+            matched_tags.add(selection)
+
+        if not matched_tags:
+            logger.warning(f"Expression '{tag_selection}' doesn't match any models.")
```

Comment:
```suggestion
            logger.warning(f"Expression 'tag:{tag_selection}' doesn't match any models.")
```
Sorry, I've just realized that you strip the `tag:` prefix upstream.